engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -289,13 +313,23 @@ class EngineJobExecutionActor(replyTo: ActorRef,
       stay
   }
 
+  private def publishHashesToMetadata(maybeHashes: Option[Try[CallCacheHashes]]) = maybeHashes match {
+    case Some(Success(hashes)) =>
+      val hashMap = hashes.hashes.collect({
+        case HashResult(HashKey(useInCallCaching, keyComponents), HashValue(value)) if useInCallCaching => (callCachingHashes + keyComponents.mkString(MetadataKey.KeySeparator.toString)) -> value
+      }).toMap
+      writeToMetadata(hashMap)
+    case _ => log.error("Cannot find hashes although they were successfully written to the database.")
```

Comment:
Hmm, different structure for success/failure sounds undesirable, but it definitely would be good to surface that "because..." *somewhere*...